src/main/scala/org/phenoscape/kb/Main.scala

```diff
@@ -151,9 +151,10 @@ object Main extends HttpApp with App {
                 }
               } ~
               path("label") {
-                parameters('iri.as[IRI]) { (iri) =>
+                parameters('iri.as[Seq[IRI]]) { (iri) =>
                   complete {
-                    Term.computedLabel(iri)
+                    val res = Term.labels(iri: _*)
+                    res.map(_.apply(0))
```

Comment:
A few issues here:
- Instead of `res`, use a descriptive variable name.
- There is a better way to get the first item; instead of `res.map(_.apply(0))`, I would use `res.map(_.head)`.
- However, if the sequence inside `res` is empty, it will throw an exception. So never use `head` unless it is impossible for the sequence to be empty. Instead use `headOption`.
- Then you have to do something to handle a possible `None` case of the option.

Reply:
@hlapp What do you want us to return in case the iri does not have a label? Should we throw an error, or just return the iri as the label, or return an empty string?

Reply:
Missing data is not an error, so don't throw an error. An empty string is not a missing value, so that's a poor choice too. If you can return a value representative if "nothing" (JS null, for example), then that should be OK. Right now in RPhenoscape I look for IRIs missing from the table returned by `/term/labels`, and then try those with `/term/label`. I assume the future behavior would be that if `/term/labels` doesn't return a label, `/term/label` won't either, because they are the same underlying implementation. So then if possible return null if no label exists.